api-channel-issue-tracker/src/main/java/com/synopsys/integration/alert/api/channel/issue/IssueTrackerProcessor.java

```diff
@@ -20,25 +20,34 @@
 import com.synopsys.integration.alert.processor.api.extract.model.project.ProjectMessage;
 
 public class IssueTrackerProcessor<T extends Serializable> {
+    private final IssueTrackerChannelLock issueTrackerLock;
     private final IssueTrackerModelExtractor<T> modelExtractor;
     private final IssueTrackerMessageSender<T> messageSender;
 
-    public IssueTrackerProcessor(IssueTrackerModelExtractor<T> modelExtractor, IssueTrackerMessageSender<T> messageSender) {
+    public IssueTrackerProcessor(IssueTrackerChannelLock issueTrackerLock, IssueTrackerModelExtractor<T> modelExtractor, IssueTrackerMessageSender<T> messageSender) {
+        this.issueTrackerLock = issueTrackerLock;
         this.modelExtractor = modelExtractor;
         this.messageSender = messageSender;
     }
 
     public final IssueTrackerResponse<T> processMessages(ProviderMessageHolder messages, String jobName) throws AlertException {
         List<IssueTrackerIssueResponseModel<T>> issueResponseModels = new LinkedList<>();
 
-        IssueTrackerModelHolder<T> simpleMessageHolder = modelExtractor.extractSimpleMessageIssueModels(messages.getSimpleMessages(), jobName);
-        List<IssueTrackerIssueResponseModel<T>> simpleMessageResponseModels = messageSender.sendMessages(simpleMessageHolder);
-        issueResponseModels.addAll(simpleMessageResponseModels);
-
-        for (ProjectMessage projectMessage : messages.getProjectMessages()) {
-            IssueTrackerModelHolder<T> projectMessageHolder = modelExtractor.extractProjectMessageIssueModels(projectMessage, jobName);
-            List<IssueTrackerIssueResponseModel<T>> projectMessageResponseModels = messageSender.sendMessages(projectMessageHolder);
-            issueResponseModels.addAll(projectMessageResponseModels);
+        try {
```

Comment:
For an issue tracker event handler it will call the processor.  The processor needs to ensure that all the creation requests are handled together so that the next event handler can search and find any created issues in order to update them and not attempt to create a duplicate issue.